pdns/nameserver.cc

```diff
@@ -301,7 +303,26 @@ bool UDPNameserver::receive(DNSPacket& packet, std::string& buffer)
     packet.d_dt.setTimeval(recvtv);
   }
   else
-    packet.d_dt.set(); // timing    
+    packet.d_dt.set(); // timing
+
+  if (g_proxyProtocolACL.match(remote)) {
+    ComboAddress psource, pdestination;
+    bool proxyProto, tcp;
+    std::vector<ProxyProtocolValue> ppvalues;
+
+    ssize_t used = parseProxyHeader(std::string(buffer, 0, len), proxyProto, psource, pdestination, tcp, ppvalues);
```

Comment:
I wonder if resizing `buffer` to `len` and then passing it directly wouldn't be easier. We will need to resize it to `DNSPacket::s_udpTruncationThreshold` before using it again anyway, because of the call to `buffer.erase(0, used)` below.

Reply:
good call on resize before use! pushed